tests/target/single-line-if-else.rs

```diff
@@ -0,0 +1,46 @@
+// rustfmt-single_line_if_else: true
+
+// Format if-else expressions on a single line, when possible.
+
+fn main() {
+    let a = if 1 > 2 { unreachable!() } else { 10 };
+
+    let a = if x {
+        1
+    } else if y { 2 } else { 3 };
+
+    let b = if cond() {
+        5
+    } else {
+        // Brief comment.
+        10
+    };
+
+    let c = if cond() {
+        statement();
+
+        5
+    } else {
+        10
+    };
+
+    if cond() {
+        statement();
+    } else {
+        other_statement();
+    }
+
+    if true {
+        do_something()
+    }
```

Comment:
I think no, it's an exceptional case: this works only if ´do_something` returns `()`, so it's basically (at human understanding level) an instruction and not an expression. (Probably the ; was forgotten in a majority of such cases.)